### PR TITLE
[TChannel part0] Add tchannel test backend

### PR DESCRIPTION
**Pull request by ChuntaoLu**
This PR adds the tchannel test backend to facilitate testing endpionts backend by tchannel downstream.

r: @Raynos

### CODE REVIEW

File: test/lib/test_backend/test_tchannel_backend.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+// Close closes the underlying channel
+func (backend *TestTChannelBackend) Close() {
+	backend.Channel.Close()
+}
+
+// Wait keeps backend process live
+func (backend *TestTChannelBackend) Wait() {
+	backend.WaitGroup.Wait()
```
Comment (Raynos) [inline test/lib/test_backend/test_tchannel_backend.go]: We do not need this `WaitGroup`, or at least tchannel server doesn't have a way to wait on the server.

Comment (ChuntaoLu) [inline test/lib/test_backend/test_tchannel_backend.go]: right, I thought we need it for the usage as https://github.com/uber/zanzibar/blob/master/benchmarks/benchserver/main.go#L52

Comment (Raynos) [inline test/lib/test_backend/test_tchannel_backend.go]: I see, that only makes sense if somebody calls `.Add(1)` and `.Done()` 

`.Wait()` immediately returns if no-one calls `.Add()`

Comment (ChuntaoLu) [inline test/lib/test_backend/test_tchannel_backend.go]: Done

**Pull request merged**